apps/src/Sounds.js

```diff
@@ -328,6 +375,71 @@ Sounds.prototype.playURL = function(url, playbackOptions) {
   }
 };
 
+/**
+ * @param {Promise} promise the promise that will resolve to the ArrayBuffer bytes to be played
+ * @param {object} playbackOptions config for the playing of the sound.
+ * @param {object} cacheParams optional parameters for caching the result of promise
+ */
+Sounds.prototype.addPromiseToSpeechQueue = function(
+  promise,
+  playbackOptions,
+  cacheParams = undefined
+) {
+  this.speechQueue.push({
+    promise: promise,
+    playbackOptions: playbackOptions,
+    cacheParams: cacheParams
+  });
```

Comment:
nit: This simplifies to `this.speechQueue.push({promise, playbackOptions, cacheParams})` using object initializer shorthand (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer)